src/Microsoft.AspNet.Razor.Runtime/TagHelpers/HtmlElementNameAttribute.cs

```diff
@@ -45,6 +47,29 @@ public HtmlElementNameAttribute([NotNull] string tag, [NotNull] params string[]
         /// <summary>
         /// An <see cref="IEnumerable{string}"/> of tag names for the <see cref="TagHelper"/> to target.
         /// </summary>
-        public IEnumerable<string> Tags { get; private set; }
+        public IEnumerable<string> Tags
```

Comment:
- go back to auto-property and do validation in the constructors, before they create lists and so on
- nit: remove `private set;`